examples/cmake/FindLibtorrentRasterbar.cmake

```diff
@@ -128,13 +128,23 @@ else()
 	if (LibtorrentRasterbar_FOUND AND NOT TARGET LibtorrentRasterbar::torrent-rasterbar)
 		add_library(LibtorrentRasterbar::torrent-rasterbar SHARED IMPORTED)
 
+		# LibtorrentRasterbar_DEFINITIONS var contains a mix of -D, -f, and possible -std options
+		# let's split them into definitions and options (that are not definitions)
+		set(LibtorrentRasterbar_defines "${LibtorrentRasterbar_DEFINITIONS}")
+		set(LibtorrentRasterbar_options "${LibtorrentRasterbar_DEFINITIONS}")
+		list(FILTER LibtorrentRasterbar_defines INCLUDE REGEX "(^|;)-D.+")
+		list(FILTER LibtorrentRasterbar_options EXCLUDE REGEX "(^|;)-D.+")
+		# remove '-D' from LibtorrentRasterbar_defines
+		string(REGEX REPLACE "(^|;)(-D)" "\\1" LibtorrentRasterbar_defines "${LibtorrentRasterbar_defines}")
```

Comment:
this all looks very GCC specific. I suppose msvc (or any non GCC-compatible compilers) don't run on a system that has `pkgconfig` (or does it?). Is matching `-D` (like a caveman) really the best/only way to know what's a preprocessor definition and not?

Reply:
I guess *we* generate this string, and we use `-D` prefixes for the `pkgconfig` script

Reply:
That's correct, although there might be a straightforward way to learn those switches (-I and -D) for the current compiler.

Reply:
I don't know where can I find an overview of command line switches for existing compilers, but checking CMake sources show that for all known to CMake C/C++ compilers (ARMCC, Bruce, Cray, GNU, IAR, Intel, NVIDIA, PGI, PathScale, SunPro, TinyCC, XL) it uses `-D`.